core/src/main/java/apoc/trigger/TriggerHandlerWrite.java

```diff
@@ -0,0 +1,153 @@
+package apoc.trigger;
+
+import apoc.SystemLabels;
+import apoc.SystemPropertyKeys;
+import apoc.util.Util;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.ResourceIterator;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.internal.helpers.collection.Pair;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import static apoc.ApocConfig.APOC_TRIGGER_ENABLED;
+import static apoc.ApocConfig.apocConfig;
+
+public class TriggerHandlerWrite {
+    public static final String NOT_ENABLED_ERROR = "Triggers have not been enabled." +
+            " Set 'apoc.trigger.enabled=true' in your apoc.conf file located in the $NEO4J_HOME/conf/ directory.";
+
+    public static Map<String, Object> toTriggerInfo(Node node) {
+        return node.getAllProperties()
+                .entrySet().stream()
+                .filter(e -> !List.of(SystemPropertyKeys.name.name(), SystemPropertyKeys.database.name()).contains(e.getKey()))
+                .collect(HashMap::new, // workaround for https://bugs.openjdk.java.net/browse/JDK-8148463
+                        (mapAccumulator, e) -> {
+                            Object value = List.of(SystemPropertyKeys.selector.name(), SystemPropertyKeys.params.name()).contains(e.getKey()) 
+                                    ? Util.fromJson((String) e.getValue(), Map.class) 
+                                    : e.getValue();
+                            
+                            mapAccumulator.put(e.getKey(), value);
+                    }, HashMap::putAll);
+    }
+
+    private static boolean isEnabled() {
+        return apocConfig().getBoolean(APOC_TRIGGER_ENABLED);
+    }
+
+    public static void checkEnabled() {
+        if (!isEnabled()) {
+            throw new RuntimeException(NOT_ENABLED_ERROR);
+        }
+    }
+
+    public static Map<String, Object> install(String databaseName, String triggerName, String statement, Map<String,Object> selector, Map<String,Object> params) {
+        checkEnabled();
```

Comment:
If this class assumes it's being executed against the system database (either because the user open a session against "system" or because Fabric routed to "system") then should we be asserting this assumption and throwing an exception when it's not the case?

Bonus points if we can also assert that it's the leader of the system database.

Reply:
PS. I quite like the idea that this should be able to work even with Fabric disabled, so long as the user is manually opening a session against "system" himself.

Reply:
I just joined the two checks, in `Trigger.java`

Reply:
Nicely done!